test/e2e/ubernetes_lite.go

```diff
@@ -61,8 +62,87 @@ var _ = framework.KubeDescribe("Multi-AZ Clusters", func() {
 	It("should schedule pods in the same zones as statically provisioned PVs", func() {
 		PodsUseStaticPVsOrFail(f, (2*zoneCount)+1, image)
 	})
+
+	It("should only be allowed to provision PDs in zones where nodes exist", func() {
```

Comment:
We can also control the zones that volumes are provisioned in, if we [name](https://github.com/kubernetes/kubernetes/blob/52ac2652bc6b62f9182cc8cd8d2365ae04794aeb/pkg/volume/util.go#L298) them correctly.  So maybe a test case we can add is to create PVCs with specific names, and make sure they get spread across the appropriate zones.

Reply:
Should this test case be added in this same PR or should I create a new PR/issue for it?

Reply:
I would put it as part of this test, maybe instead of (or in addition to) comparing the GetAllZones call.  Because from a user's standpoint, they would see issues in the volume provisioning.  The GetAllZones call is just an implementation detail.

Reply:
I added it in addition to. So the GetAllZones check still exists, the PVC checking follows.